### PR TITLE
Fix receiving of downstream messages

### PR DESCRIPTION
**Pull request by gregersrygg**
Turning off power save mode have a bug that I have a fix for in a separate PR, so the example won't work without that fix.

### CODE REVIEW

**Comment by gregersrygg** [comment]
Merging now, but please review anyways.

File: TelenorNBIoT.cpp [hunk 1]
```diff
@@ -369,9 +375,13 @@ bool TelenorNBIoT::receiveFrom(char *ip, uint16_t *port, char *outbuf, uint16_t
             {
                 *port = atoi(fields[2]);
             }
+            
             *length = atoi(fields[3]);
-            // TODO(stalehd): Convert data into bytes
-            strcpy(outbuf, fields[4]);
+            char *pos = fields[4];
+            pos++; // inc by 1 to skip double-quote char
+            hexToBytes(pos, *length, outbuf);
```
Comment (gordonklaus) [inline TelenorNBIoT.cpp]: This isn't a new issue, but `outbuf` needs to be big enough to hold all `*length` bytes.  Maybe `length` should also be an input parameter and should hold the size of `outbuf`, so we don't write beyond its allocation.

File: TelenorNBIoT.cpp [hunk 2]
```diff
@@ -425,21 +435,43 @@ bool TelenorNBIoT::isError(const char *line)
 
 int TelenorNBIoT::splitFields(char *line, char **fields)
 {
-    int found = 0;
+    int found = 1;
     char *p = line;
     fields[0] = line;
-    while (*p++)
+    while (*++p)
```
Comment (gordonklaus) [inline TelenorNBIoT.cpp]: You were right to think we should avoid this style of incrementing, even though it is idiomatic.  It is clearer to do the increment separate from the check.

File: TelenorNBIoT.cpp (hunk 1 above)
Comment (gregersrygg) [reply TelenorNBIoT.cpp]: Suggested fix in #14

**Pull request merged**